src/v/cluster/tm_stm.cc

```diff
@@ -550,22 +550,25 @@ ss::future<tm_stm::op_status> tm_stm::add_partitions(
     if (ptx->second.status != tm_transaction::tx_status::ongoing) {
         co_return tm_stm::op_status::unknown;
     }
-    bool just_started = ptx->second.partitions.size() == 0
-                        && ptx->second.groups.size() == 0;
 
-    if (just_started) {
-        tm_transaction tx = ptx->second;
-        for (auto& partition : partitions) {
-            tx.partitions.push_back(partition);
-        }
-        tx.last_update_ts = clock_type::now();
-        auto r = co_await update_tx(tx, tx.etag);
+    if (!is_transaction_ga()) {
+        bool just_started = ptx->second.partitions.size() == 0
+                            && ptx->second.groups.size() == 0;
 
-        if (!r.has_value()) {
-            co_return tm_stm::op_status::unknown;
+        if (just_started) {
```

Comment:
I didn't understand the intent of this change.

Reply:
added commit message